### DISCUSSION TITLE
Car Navigation Integration

### DISCUSSION THREAD

**Discussion by Jimbo757767** [BimmerGestalt/AAIdrive#494] [General]
I've just installed 1.3.1 with the screen mirroring addon. Great. With Google maps running in landscape mode it follows the route OK without much lag. 
A few things I don't get from the implemented features section of the overview:
_Android Navigation Intents and buttons can be handled by the connected car's navigation_
What does this do? I found I could stop Google Maps Nav from the car screen, but that's all. Can I input or change things using the car controls?
_Google Maps can share destinations to the connected car's navigation_
I can't find anything in Google maps to do this
_Addresses can be entered in the phone interface to start the car's navigation_
I found I can only input an address when AAIdrive is connected to the car. It would be useful to input an address before linking so that it would navigate as soon as connecting.
It's great work

**Comment by hufman** [comment]
I'm happy you're enjoying it! The car navigation integration is unrelated to the screen mirroring addon.

Android Navigation Intents refers to the various Navigation buttons in other apps, like a car charger map app or a mobile food ordering app, and some phones can even trigger navigation after long-pressing on a street address in a webpage. In the chooser menu that shows up, alongside any installed navigation apps such as Google Maps and Waze, the car navigation will show up while it is connected.

Google Maps place search, however, is hardcoded to start navigation in Google Maps itself. Instead, you must Share a Place from Google Maps, and then the car will show up as a destination that can be shared to.

Thank you for the suggestion! Currently the app doesn't save any destinations that are passed through to the car, but I could see how it would integrate with an old feature idea to save a list of favorite destinations #4
